packages/kbn-apm-synthtrace/src/lib/apm/instance.ts

```diff
@@ -22,6 +23,45 @@ export class Instance extends Entity<ApmFields> {
     });
   }
 
+  dbExitSpan({ spanName, subType }: { spanName: string; subType?: string }) {
+    const exitSpan = new Span({
+      ...this.fields,
+      'service.target.type': subType,
+      'span.destination.service.name': '', //deprecated
+      'span.destination.service.resource': subType,
+      'span.destination.service.type': '', //deprecated
+      'span.name': spanName,
+      'span.subtype': subType,
+      'span.type': 'db',
+    });
+
+    return exitSpan;
+  }
+
+  messagingExitSpan(mesageQueueName: string) {}
+
+  httpExitSpan({ spanName, destination }: { spanName: string; destination: string }) {
```

Comment:
does this need to be a method on the instance? can it be a function that returns the correct properties to be passed to `.span()`?

Reply:
Currently span only take name, type and subtype. What signature do you suggest?

Reply:
`span ( spanName:string, spanType:string, spanSubtype:string, fields?:ApmFields )`

and then something like:

```ts
function httpExitSpan ( options:{ spanName:string; destination:string } ) {
	return [ spanName, 'ext', 'http', { 'span.destination.service.resource': destination } ];
}

instance.span(...httpExitSpan({ spanName: 'GET /api/products', destination: 'opbeans:3000' }));
```

Reply:
I can change span to:

```ts
   span(spanFields: ApmFields & { ['span.name']: string; ['span.type']: string }) {
    return new Span({ ...this.fields, ...spanFields });
  }
```

Then call it like:

```ts
opbeansRum
  .span(getHttpExitSpan({
    spanName: 'GET /api/products/top',
    destination: 'http://opbeans-node:3000',
  })
  .duration(300)
  .timestamp(timestamp)
```

I'm fine with that. WDYT?

Reply:
But what's the problem with an instance method. Don't you think it looks cleaner than `span(...httpExitSpan(` ?

Reply:
@sqren I think it'd be good to avoid having to use the ES field names in often-used APIs, it requires a lot of knowledge about the underlying schema. (e.g. I think `spanName` is better than `span.name`). Other than that I think it looks good.